miner/block_builder.go

```diff
@@ -278,29 +268,47 @@ func (t *BlockBuilder) listenForTx() {
 				// not accepting txs when not synced
 				continue
 			}
-			if data != nil {
-
-				x, err := types.BytesAsSignedTransaction(data.Bytes())
-				if err != nil {
-					t.Log.Error("cannot parse incoming TX")
-					continue
-				}
+			if data == nil {
+				continue
```

Comment:
This happens when the channel is closed during graceful shutdown. Maybe the correct behaviour should be to print a message ("Ending transaction listening") and `return`ing instead of `continue`ing.